lib/src/parser.dart

```diff
@@ -0,0 +1,222 @@
+import 'dart:convert';
+
+import 'proto/insertion_order.pb.dart';
+
+class Parser {
+  /// Parses a json string to [InsertionOrder].
+  static InsertionOrder ioFromJson(String jsonString) {
+    Map<String, dynamic> map = json.decode(jsonString);
+    return createInsertionOrder(map);
+  }
+
+  /// Creates an [InsertionOrder] instance from [map].
+  ///
+  /// Throws an [FormatException] if a required field is missing.
+  static InsertionOrder createInsertionOrder(Map<String, dynamic> map) {
+    final insertionOrder = InsertionOrder();
+
+    if (!map.containsKey('advertiserId')) throw FormatException;
```

Comment:
Rather than throwing, I generally prefer 'reasonable defaults'. For something like this it doesn't have to be 'all or nothing'.
You can fill in the IO like this:
  final insertionOrder = InsertionOrder()
      ..advertiserId = map['advertiserId'] ?? 'Unknown'
      ..campaignId = map['campaignId'] ?? 'Unknown'
     .... and so on